google/resource_logging_project_exclusion_test.go

```diff
@@ -0,0 +1,225 @@
+				),
+			}, {
+				Config: testAccLoggingProjectExclusion_basicUpdated(exclusionName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectExclusionExists("google_logging_project_exclusion.basic", &exclusionAfter),
+					testAccCheckLoggingProjectExclusion(&exclusionAfter, "google_logging_project_exclusion.basic"),
+				),
+			},
+		},
+	})
+
+	// Filter should have changed, but Description and Disabled should be the same
+	if exclusionBefore.Description != exclusionAfter.Description {
+		t.Errorf("Expected Description to be the same, but it differs: before = %#v, after = %#v",
+			exclusionBefore.Description, exclusionAfter.Description)
+	}
+	if exclusionBefore.Filter == exclusionAfter.Filter {
+		t.Errorf("Expected Filter to change, but it didn't: Filter = %#v", exclusionBefore.Filter)
+	}
+	if exclusionBefore.Disabled != exclusionAfter.Disabled {
+		t.Errorf("Expected Disabled to be the same, but it differs: before = %#v, after = %#v",
+			exclusionBefore.Disabled, exclusionAfter.Disabled)
+	}
+}
+
+func testAccCheckLoggingProjectExclusionDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_logging_project_exclusion" {
+			continue
+		}
+
+		attributes := rs.Primary.Attributes
+
+		_, err := config.clientLogging.Projects.Exclusions.Get(attributes["id"]).Do()
+		if err == nil {
+			return fmt.Errorf("project exclusion still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLoggingProjectExclusionExists(n string, exclusion *logging.LogExclusion) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		attributes, err := getResourceAttributes(n, s)
+		if err != nil {
+			return err
+		}
+		config := testAccProvider.Meta().(*Config)
+
+		si, err := config.clientLogging.Projects.Exclusions.Get(attributes["id"]).Do()
+		if err != nil {
+			return err
+		}
+		*exclusion = *si
+
+		return nil
+	}
+}
+
+func testAccCheckLoggingProjectExclusion(exclusion *logging.LogExclusion, n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		attributes, err := getResourceAttributes(n, s)
+		if err != nil {
+			return err
+		}
+
+		if exclusion.Description != attributes["description"] {
+			return fmt.Errorf("mismatch on description: api has %s but client has %s", exclusion.Description, attributes["description"])
+		}
+
+		if exclusion.Filter != attributes["filter"] {
+			return fmt.Errorf("mismatch on filter: api has %s but client has %s", exclusion.Filter, attributes["filter"])
+		}
+
+		disabledAttribute, err := toBool(attributes["disabled"])
+		if err != nil {
+			return err
+		}
+		if exclusion.Disabled != disabledAttribute {
+			return fmt.Errorf("mismatch on disabled: api has %t but client has %t", exclusion.Disabled, disabledAttribute)
+		}
+
+		return nil
+	}
+}
+
+func toBool(attribute string) (bool, error) {
+	// Handle the case where an unset value defaults to false
```

Comment:
nit: This can be made a bit simpler:
```
if attribute == "" {
  return false, nil
}
return strconv.ParseBool(attribute)
```
The error message that ParseBool returns for an invalid param is `strconv.ParseBool: parsing "not-a-bool": invalid syntax` which I think is plenty descriptive, especially given that it's just a test failure (as opposed to an end-user message)